components/script/dom/htmlscriptelement.rs

```diff
@@ -669,7 +830,9 @@ impl HTMLScriptElement {
         // unminified content.
         let (input, output) = (tempfile::NamedTempFile::new(), tempfile::tempfile());
         if let (Ok(mut input), Ok(mut output)) = (input, output) {
-            input.write_all(script.text.as_bytes()).unwrap();
+            if let SourceCode::Text(text) = &script.code {
+                input.write_all(text.as_bytes()).unwrap();
+            }
```

Comment:
Here I think we should also warn in the else case(or find a way to get the text from a `JSScript`)

Reply:
maybe we can use `JS_DecompileScript `method with uses `context` and `Handle<*mut JSScript>` and returns `*mut JSString`? I am unsure on how to get the context though.

Reply:
Good idea, give it a try I would say. You can get the global from(the equivalent of) `*self.globa().get_cx()`

Reply:
I noticed that the wrapper for  [`DecompileScript`](https://docs.rs/mozjs/0.7.2/mozjs/rust/wrappers/fn.JS_DecompileScript.html) takes 4 args, while the [one](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_reference/JS_DecompileScript) in spidermonkey project, takes 2 and [JS_DecompileScriptObject](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_reference/JS_DecompileScriptObject) takes 4. Due to lack of documentation on the last one, not sure if we can call it.

Reply:
I think you can try to use this one: https://github.com/servo/mozjs/blob/aabcc9ba889b2755f1e4e83f28323a60415a790f/mozjs/js/src/jsapi.h#L1885

Reply:
We could also just store a clone of the original text alongside the compiled script, and use that when needed.

Reply:
Yeah, that's always an option. I think it would be better, as decompile script doesnt always return the original script.

Reply:
yes.

Reply:
So I tried to store it as DOMString, but Send is not implemented for it and cannot queue the task back. Should I implement that similar to how you did for OffThreadToken?

Reply:
I think you should store it as a String then, and later convert it back to a dom string inside the task.